test/integration/errors-oidc.js

```diff
@@ -0,0 +1,98 @@
+const supertest = require('supertest')
+const ldnode = require('../../index')
+const path = require('path')
+const fs = require('fs-extra')
+const expect = require('chai').expect
+
+describe('OIDC error handling', function () {
+  process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0'
+
+  const serverUri = 'https://localhost:3457'
+  var ldpHttpsServer
+  const rootPath = path.join(__dirname, '../resources/accounts/errortests')
+  const dbPath = path.join(__dirname, '../resources/accounts/db')
+
+  const ldp = ldnode.createServer({
+    root: rootPath,
+    sslKey: path.join(__dirname, '../keys/key.pem'),
+    sslCert: path.join(__dirname, '../keys/cert.pem'),
+    auth: 'oidc',
+    webid: true,
+    idp: false,
+    strictOrigin: true,
+    dbPath,
+    serverUri
+  })
+
+  before(function (done) {
+    ldpHttpsServer = ldp.listen(3457, done)
+  })
+
+  after(function () {
+    if (ldpHttpsServer) ldpHttpsServer.close()
+    fs.removeSync(path.join(rootPath, 'index.html'))
+    fs.removeSync(path.join(rootPath, 'index.html.acl'))
+  })
+
+  const server = supertest(serverUri)
+
+  describe('Unauthenticated requests to protected resources', () => {
+    describe('accepting text/html', () => {
+      it('should return 401 Unauthorized with www-auth header', () => {
+        return server.get('/profile/')
+          .set('Accept', 'text/html')
+          .expect('WWW-Authenticate', 'Bearer realm="https://localhost:3457", scope="openid"')
+          .expect(401)
+      })
+
+      it('should return an html redirect body', () => {
+        return server.get('/profile/')
+          .set('Accept', 'text/html')
+          .expect('Content-Type', 'text/html; charset=utf-8')
+          .then(res => {
+            expect(res.text).to.match(/<meta http-equiv="refresh" content="0; url=https:\/\/localhost:3457\/api\/auth\/select-provider/)
+          })
+      })
+    })
+
+    describe('not accepting html', () => {
+      it('should return 401 Unauthorized with www-auth header', () => {
+        return server.get('/profile/')
+          .set('Accept', 'text/plain')
+          .expect('WWW-Authenticate', 'Bearer realm="https://localhost:3457", scope="openid"')
+          .expect(401)
+      })
+    })
+  })
+
+  describe('Authenticated responses to protected resources', () => {
+    describe('with an empty bearer token', () => {
```

Comment:
Huh. According to [section 3.1](https://tools.ietf.org/html/rfc6750#section-3.1) of the spec, a missing bearer token seems to satisfy the `invalid_request`/`400` scenario:

>    invalid_request
         The request is missing a required parameter, includes an
         unsupported parameter or parameter value, repeats the same
         parameter, uses more than one method for including an access
         token, or is otherwise malformed.  The resource server SHOULD
         respond with the HTTP 400 (Bad Request) status code.